codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/AwsServiceModel.java

```diff
@@ -509,7 +510,7 @@ private void addCasesForMember(MethodSpec.Builder methodBuilder, MemberModel mem
 
         result.add(memberGetter(member));
 
-        return result.stream();
+        return checkDeprecated(member, result).stream();
```

Comment:
I struggled a little bit with the decision on where to implement the `checkDeprecated` logic. Ideally it might make more sense to be defined at the `MemberModel` level, but I could not find a common entry or exit point that they could all safely share, and there exists other logic that will do something like `MemberModel#getDocuemtnation(..)` and then append more to the end, when it seems we want to guarantee `@deprecated` is consistently at the end. Ideas/opinions welcome.

Reply:
I actually prefer it the way you have it; we should keep the model objects as dumb as possible